### ISSUE TITLE
Using camera plugin with fast_qr_reader_view plugin

### ISSUE THREAD

**Issue by c0h1b4** [flutter/flutter#26548]
I am developing an app to read QR codes and then take a photo of an object. For the QR reading I'm using the fast_qr_reader_view plugin.
When I add the camera plugin (dependency), I get a lot of warnings and several errors from the XCode compiler. specifically those:

```
=== BUILD TARGET camera OF PROJECT Pods WITH CONFIGURATION Debug ===
error: the following command failed with exit code 0 but produced no further output
CompileC /Users/mauro/Library/Developer/Xcode/DerivedData/Runner-alpqoqfbegkriafmeocxxerreajl/Build/Intermediates.noindex/Pods.build/Debug-iphoneos/camera.build/Objects-normal/armv7/CameraPlugin.o /Users/mauro/development/flutter/.pub-cache/hosted/pub.dartlang.org/camera-0.2.9+1/ios/Classes/CameraPlugin.m normal armv7 objective-c com.apple.compilers.llvm.clang.1_0.compiler
duplicate symbol _OBJC_IVAR_$_FLTCam._captureVideoOutput in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._videoWriterInput in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._captureVideoInput in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._assetWriterPixelBufferAdaptor in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._videoWriter in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._latestPixelBuffer in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._captureSession in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_CLASS_$_FLTCam in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_METACLASS_$_FLTCam in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._eventChannel in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._eventSink in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._previewSize in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._captureSize in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._onFrameAvailable in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._captureDevice in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
duplicate symbol _OBJC_IVAR_$_FLTCam._textureId in:
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/camera/libcamera.a(CameraPlugin.o)
/Users/mauro/development/projects/lab1click/build/ios/Debug-iphoneos/fast_qr_reader_view/libfast_qr_reader_view.a(FastQrReaderViewPlugin.o)
ld: 16 duplicate symbols for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
Could not build the precompiled application for the device.`
```
Any ideas to use both plugins on the same project?

Thanks
David

**Comment by c0h1b4** [comment]
IMHO when the dependency (plugin) uses the same exposed variables as other plugins we get this conflicts. Would there be a way to encapsulate the plugins from each other?

**Comment by kangwang1988** [comment]
@c0h1b4 
I've checked the issue, that's because developers define same class `FLTCam` in CameraPlugin.m(from plugin camera) and FastQrReaderViewPlugin.h(from fast_qr_reader_view).
I do think this is the responsibility of plugin developers to avoid this, like add customized prefix for their class and methods, etc. Otherwise, the developer who want to use them both will have to do this.
Closing #26548
